mantidproject/mantid
# Fix offset problem in CreateSampleWorkspace

**Description of work.**

The code in CreateSampleWorkspace evaluates the user-specified function at the left hand edge of each bin and then writes the resulting y value into the bin. This is misleading and results in functions which are symmetric about 0 appearing slightly asymmetric in the resulting workspace

The unit test\system tests runs in Jenkins have highlighted the fact that the FindEPP algorithm included some logic which was perhaps driven by the old code in CreateSampleWorkspace. It also used the left hand bin edge of the bin with the maximum y to set the initial fit parameters on a peak fit. I think this was incorrect so have changed it. The algorithm was written by the ILL team so will check with them that they're happy with this change.

**To test:**

Run this script and check that the y values in the first and last bin are the same:
```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

function = "name=Lorentzian,Amplitude=1,PeakCentre=5,FWHM=1"
ws = CreateSampleWorkspace("Histogram", Function="User Defined", UserDefinedFunction=function, 
                            XMin=-5, XMax=5, BinWidth=0.01, XUnit="DeltaE")
```

*There is no associated issue.*



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
